src/java.base/share/classes/jdk/internal/foreign/abi/x64/windows/CallArranger.java

```diff
@@ -164,7 +162,7 @@ VMStorage nextStorage(int type, MemoryLayout layout) {
             if (nRegs >= MAX_REGISTER_ARGUMENTS) {
                 assert forArguments : "no stack returns";
                 // stack
-                long alignment = Math.max(SharedUtils.alignment(layout, true), STACK_SLOT_SIZE);
+                long alignment = Math.max(layout.byteAlignment(), STACK_SLOT_SIZE);
```

Comment:
Actually, this patch is missing most of the stuff in PR 728. I was under the impression that, in order to fully support that, some VM changes were needed (e.g. to have better granularity in call shuffling - as per https://github.com/openjdk/panama-foreign/pull/699). As a result, this PR only contains changes to SharedUtil (to remove unused alignment functions) - but nothing else.

Reply:
699 is not needed for this. 728 is a pure Java change that simply rejects layouts that don't have their natural alignment (so, it will rejects packed structs for instance, since the implementation doesn't support them on all platforms). All the other changes from 728 are here (most notably the code in AbstractLinker that checks the alignment), except the change that ignores the `layout` here and turns the code around the line above into an `assert`.

The mac stack spilling patch requires 699 though (https://github.com/openjdk/panama-foreign/pull/746). I will put that in the PR with the VM changes.

Reply:
Thanks for the clarification - I will incorporate those changes as well then.